homeassistant/components/shelly/coordinator.py

```diff
@@ -140,7 +145,7 @@ def async_setup(self) -> None:
             model=MODEL_NAMES.get(self.model, self.model),
             sw_version=self.sw_version,
             hw_version=f"gen{get_device_entry_gen(self.entry)} ({self.model})",
-            configuration_url=f"http://{self.entry.data[CONF_HOST]}",
+            configuration_url=f"http://{self.entry.data[CONF_HOST]}:{self.entry.data.get(CONF_PORT, DEFAULT_HTTP_PORT)}",
```

Comment:
`get_http_port`

Need to add a test that verify that existing entry that doesn't have the port in stored config entry is loaded correctly, otherwise someone might remove this get in the future thinking it is not needed.

Reply:
Sorry but I miss the point: we load a default for all config entry that doesn't have a port.

Reply:
Correct, but we don't have any test that actually validates what port we pass to `aioshelly` and that if we don't have a port we pass the default one. Basically besides 1 config flow test (which doesn't call `aioshelly`) there are no tests to cover the port usage. Should be something similar to https://github.com/home-assistant/core/pull/110997#issuecomment-1987393418